src/R2DH_alg.jl

```diff
@@ -0,0 +1,283 @@
+export R2DH
+
+"""
+    R2DH(nlp, h, options)
+    R2DH(f, ∇f!, h, options, x0)
+
+A first-order quadratic regularization method for the problem
+
+    min f(x) + h(x)
+
+where f: ℝⁿ → ℝ has a Lipschitz-continuous gradient, and h: ℝⁿ → ℝ is
+lower semi-continuous, proper and prox-bounded.
+
+About each iterate xₖ, a step sₖ is computed as a solution of
+
+    min  φ(s; xₖ) + ψ(s; xₖ)
+
+where φ(s ; xₖ) = f(xₖ) + ∇f(xₖ)ᵀs + ½ sᵀ (σₖ+Dₖ) s (if summation = true) and φ(s ; xₖ) = f(xₖ) + ∇f(xₖ)ᵀs + ½ sᵀ (σₖ+Dₖ) s (if summation = true) is a quadratic approximation of f about xₖ,
```

Comment:
You said 2 times if summation = true.
Add the single quote  to indicate that this is some code.
You can preview the Docs locally by running the `makedocs` function in docs/make.jl of this repo.